mantidimaging/core/operations/roi_normalisation/roi_normalisation.py

```diff
@@ -51,44 +44,80 @@ def filter_func(images: ImageStack,
         If the Air Region is in bounds, but has overlapping coordinates
         the crop give back a 0 shape of the coordinates that were wrong.
-
         :param images: Sample data which is to be processed. Expected in radiograms
-
         :param region_of_interest: The order is - Left Top Right Bottom. The air
         region for which grey values are summed up and used for normalisation/scaling.
-
         :param normalisation_mode: Controls what the ROI counts are normalised to.
             'Stack Average' : The mean value of the air region across all projections is preserved.
             'Flat Field' : The mean value of the air regions in the projections is made equal to the mean value of the
                            air region in the flat field image.
-
         :param flat_field: Flat field to use if 'Flat Field' mode is enabled.
-
         :param progress: Reference to a progress bar object
-
         :returns: Filtered data (stack of images)
         """
-        if normalisation_mode not in modes():
-            raise ValueError(f"Unknown normalisation_mode: {normalisation_mode}, should be one of {modes()}")
+        if not region_of_interest:
+            raise ValueError('region_of_interest must be provided')
 
-        if normalisation_mode == "Flat Field" and flat_field is None:
-            raise ValueError('flat_field must provided if using normalisation_mode of "Flat Field"')
+        if isinstance(region_of_interest, list):
+            region_of_interest = SensibleROI.from_list(region_of_interest)
 
         h.check_data_stack(images)
 
-        if not region_of_interest:
-            raise ValueError('region_of_interest must be provided')
+        global_params = RoiNormalisationFilter.calculate_global(images, region_of_interest, normalisation_mode,
+                                                                flat_field)
+
+        params = {
+            'region_of_interest': region_of_interest,
+            'normalisation_mode': normalisation_mode,
+            'global_params': global_params
+        }
+
+        print("Filter Function Params:", params)
+
+        ps.run_compute_func(RoiNormalisationFilter.compute_function, images.data.shape[0], images.shared_array, params)
 
-        progress = Progress.ensure_instance(progress, task_name='ROI Normalisation')
-        _execute(images, region_of_interest, normalisation_mode, flat_field, progress)
         h.check_data_stack(images)
+
         return images
 
+    @staticmethod
+    def calculate_global(images, region_of_interest, normalisation_mode, flat_field):
+        global_params = {}
+        air_means = np.array([
+            RoiNormalisationFilter._calc_mean(images.data[i], region_of_interest.left, region_of_interest.top,
+                                              region_of_interest.right, region_of_interest.bottom)
+            for i in range(images.data.shape[0])
+        ])
```

Comment:
We can do this a single numpy call:
`air_means = images.data[:, region_of_interest.top:region_of_interest.bottom, region_of_interest.left:region_of_interest.right].mean(axis=(1, 2))`

Then can remove `_calc.mean()`